src/app.tsx

```diff
@@ -13,18 +13,40 @@ html, body {
 }
 `
 
-export default class App extends React.Component {
-  constructor(props) {
+export type DisplayRepo = {
+  description: string,
+  language?: string,
+  name: string,
+  stars: number,
+  url: string,
+}
+
+type ResponseRepo = {
+  archived: boolean,
+  description: string,
+  fork: boolean,
+  html_url: string,
+  language?: string,
+  name: string,
+  stargazers_count: number,
+}
+
+type State = {
+  repos: Array<DisplayRepo>
+}
+
+export default class App extends React.Component<{}, State> {
+  constructor(props: any) {
     super(props)
     this.state = { repos: [] }
   }
 
   getFromStorage() {
     try {
-      const data = JSON.parse(localStorage.getItem(storageKey))
+      const data = JSON.parse(localStorage.getItem(storageKey) || '""')
       const now = new Date()
       const then = new Date(data.date)
-      if ((now - then) / 36e5 <= 24) {
+      if ((+now - +then) / 36e5 <= 24) {
```

Comment:
i get what this is doing, but i don't really like it

Reply:
I don't love it, but TS doesn't let you do math with plain old dates